File: php/testcode/xss_008.php
<?php
require_once __DIR__ . '/shared.php';

// vuln-code-snippet start php_xss_attribute_escaped
function xss_attribute_escaped(BenchmarkRequest $req): BenchmarkResponse {
    $input = $req->param('avatar');

    $safe = htmlspecialchars($input, ENT_QUOTES, 'UTF-8'); // vuln-code-snippet safe-line php_xss_attribute_escaped
    $html = '<img src="' . $safe . '" alt="avatar">';

    return BenchmarkResponse::html($html);
}
// vuln-code-snippet end php_xss_attribute_escaped

File: ruby/testcode/xss_008.rb
require_relative 'shared'

# vuln-code-snippet start ruby_xss_content_type_json
def xss_content_type_json(req)
  user_id = req.param('user_id')
  filter = req.param('filter', default: 'active')
  results = [
    { id: user_id, filter: filter, status: 'ok' }
  ]
  BenchmarkResponse.json(results) # vuln-code-snippet safe-line ruby_xss_content_type_json
end
# vuln-code-snippet end ruby_xss_content_type_json
